fix: extend documentation (#99)
aerogear/offix: docs/ref-conflict-client.md

# Client Side Conflict Resolution

When performing data synchronization between multiple clients it is common for remote devices to become offline for a certain amount of time. As a result of being offline, data that is modified by a client can become outdated with the server. Further operations on that record can cause a conflict (often called a collision). For more information about offline workflows, please see [Offline Support](ref-offline.md)

Offix provides a way to manage and resolve these conflicts for any GraphQL type.
Conflict implementation will require additional elements in your application in order to work:

 - Mutation `returnType` added on context to any mutation
 - Additional metadata inside types (for example version field) depending on conflict implementation 

Conflict mechanism is divided between:

- Conflict detection
Developers can detect conflicts on resolver level
- Conflict resolution 
Conflicts are sent back to client and resolved by resending data back to server.

> . Offix will allow to detect and resolve conflict without user interactions. 

Reviewer: ```suggestion
Offix allows developers to detect and resolve conflict without user interactions. 
```

> By default when no changes were made on the same fields, implementation will try to resend modified payload back to server. When changes on server and client covering the same fields one of the specified conflict resolution strategies can be used. Default strategy will apply client changes on top of the server side.

Reviewer: ```suggestion
By default when no changes were made on the same fields, the implementation will try to resend the modified payload back to the server. When changes on the server and the client cover the same fields one of the specified conflict resolution strategies can be used. The default strategy will apply client changes on top of the server side.
```

> Developers can modify strategies for their needs.

Reviewer: ```suggestion
Developers can modify strategies to suit their needs.
```

}
```

### Custom Conflict implementation by extening ObjectState

Offix enables flexibility on how conflicts are detected and resolved. 
In many cases developers may need different ways of detecting conflits than relying on version field
stored in database. For example if database already have `changedAt` field that is being supported by trigger it can be used as custom conflict implemementation.

Under the hood conflicts implementations are extending an `ObjectState` interface.
This interface exist on both client and server and provides functions that help with detection and resolution of conflicts.

> Default implementation an `VersionedObjectState`. This means Offix expects all data which could be conflicted to have a version field. If this is not the case, developers can also provide custom state which Offix will then use for conflict resolution. To do this, Offix expects certain functions to be available under the `conflictProvider` option in config. These functions and their signatures are:

Reviewer: ```suggestion
The default implementation is `VersionedObjectState`. This means Offix expects all data which could be conflicted to have a version field. If this is not the case, developers can also provide custom state which Offix will then use for conflict resolution. To do this, Offix expects certain functions to be available under the `conflictProvider` option in config. These functions and their signatures are:
```
